### PR TITLE
Forum fixes

### PR DESCRIPTION
**Pull request by allenwq**
The "RE RE RE" thing makes forum experience quite frustrated, examples here: https://beta.coursemology.org/courses/2/forums/reflections-on-lectures

This PR:
- Make post title optional and do not display post title in forum
- Validates that post content cannot be empty

Maybe we should remove the title in db next.

screen shot 2016-09-16 at 8 37 21 pm

### CODE REVIEW

File: spec/features/course/forum/post_management_spec.rb [hunk 1]
```diff
@@ -23,9 +23,9 @@
       scenario 'I can create a post' do
         visit course_forum_topic_path(course, forum, topic)
 
-        # Create a post with a missing title.
+        # Create a post with a empty content.
```
Comment (fonglh) [inline spec/features/course/forum/post_management_spec.rb]: "Create a post with empty content."

Comment (allenwq) [inline spec/features/course/forum/post_management_spec.rb]: Fixed !

**Comment by allenwq** [comment]
This helps with #1498

**Comment by weiqingtoh** [comment]
@allenwq - Can add 1 more rspec matcher - `it { should validate_presence_of(:text) }` at the post model? 

Other than that, LGTM. For title, can remove or keep it - no difference to me. Not sure if we might need the title next

**Comment by allenwq** [comment]
@weiqingtoh I used few feature specs to test the validation, do we want to duplicate it in model ?

**Comment by allenwq** [comment]
My another finding of shoulda matchers is, it only tests the declaration, not the behaviour. So it's not actually reliable. 

If we say `it is_expected.to have_many(:records).dependent(:destroy)`, the shoulda matchers will pass if you have the declaration in model, but the actual model might still fail to destroy the dependent objects due to foreign key or runtime issues.

**Comment by weiqingtoh** [comment]
Hmmm both are fine. I think the validation match is just more explicit, so if someone removes it next time, it is easier to know where it failed. 

If that's the case, I think this is good for merge.

**Comment by allenwq** [comment]
@weiqingtoh Let me add it in this case then, but think we need to keep in mind the behaviour of shoulda matchers.

**Comment by allenwq** [comment]
@weiqingtoh Updated !

**Pull request merged**